web/packages/teleterm/src/mainProcess/agentRunner/agentRunner.ts

```diff
@@ -159,7 +152,8 @@ export class AgentRunner {
 
     process.once('spawn', spawnHandler);
     process.once('error', errorHandler);
-    process.once('exit', exitHandler);
+    // Using close instead of exit to ensure stderr has been closed and we captured all logs.
+    process.once('close', closeHandler);
```

Comment:
[My original suggestion](https://github.com/gravitational/teleport/pull/29479#discussion_r1277286328) to replace `exit` with `close` was wrong.

We're never going to spawn an agent with inherited stdio, so the concern I described just isn't realistic. What's more, in the `exitHandler` we used to return logs together with the exit status. But when the exit event is fired, the stderr and stdout are not guaranteed to have flushed all the output.